installscripts/jazz-terraform-unix-noinstances/chefProvisioner.tf

```diff
@@ -109,10 +109,10 @@ resource "null_resource" "configureJenkinsDocker" {
 }
 
 resource "null_resource" "configureCliJenkins" {
-  depends_on = ["null_resource.configureJenkinsDocker", "null_resource.configureJenkinsInstance"]
+  depends_on = ["null_resource.configureJenkinsDocker", "null_resource.configureJenkinsInstance", "aws_iam_user_policy.operational_policy"]
```

Comment:
Why the explicit dependency on the policy resource? The script only appears to need the access secret key and ID, which are already dependencies because they are referenced.

Reply:
aws_iam_user_policy.operational_policy has dependency on resource "aws_iam_access_key" and it has dependency on resource "aws_iam_user". ie, IAM operational user must be created before running the docker, since it needs access key and secret key of that user.

Reply:
`since it needs access key and secret key of that user.`

If the script has a dependency on those two things, can't you pass them in as explicit arguments and depend on them directly?

Reply:
If I remove this depends_on, it will throw error,

Error: resource 'null_resource.configureCliJenkins' provisioner local-exec (#1): unknown resource 'aws_iam_access_key.operationalpolicy' referenced in variable aws_iam_access_key.operationalpolicy.secret
Error: resource 'null_resource.configureCliJenkins' provisioner local-exec (#1): unknown resource 'aws_iam_access_key.operationalpolicy' referenced in variable aws_iam_access_key.operationalpolicy.id